lib/Client.php

```diff
@@ -127,9 +128,46 @@ public function getUserInfo() {
 			return $this->getAccessTokenPayload();
 		}
 
+		if (isset($openIdConfig['use-access-token-introspection-for-user-info']) && $openIdConfig['use-access-token-introspection-for-user-info']) {
+			$introspectionClientId = $openIdConfig['token-introspection-endpoint-client-id'] ?? null;
+			$introspectionClientSecret = $openIdConfig['token-introspection-endpoint-client-secret'] ?? null;
+			$accessToken = $this->getAccessToken();
+			if (isset($openIdConfig['exchange-token-mode-before-introspection'])) {
+				$mode = $openIdConfig['exchange-token-mode-before-introspection'];
+				$token = $mode === 'refresh-token' ? $this->getRefreshToken() : $this->getAccessToken();
+				$this->logger->debug("Starting token-exchange to get user_info with subject_token mode: $mode");
+				$accessToken = $this->exchangeToken($token, $mode);
+			}
+
+			return $this->introspectToken($accessToken, '', $introspectionClientId, $introspectionClientSecret);
+		}
+
 		return $this->requestUserInfo();
 	}
 
+	/**
+	 * Perform a RFC8693 Token Exchange
+	 * https://datatracker.ietf.org/doc/html/rfc8693
+	 *
+	 * @param string $subjectToken
+	 * @param string $tokenType Type of the token to exchange 'refresh-token' or 'access-token'
+	 * @return string Access Token
+	 * @throws OpenIDConnectClientException
+	 */
+	public function exchangeToken(string $subjectToken, string $tokenType): string {
+		$subjectTokenType = $tokenType === 'refresh-token' ? 'urn:ietf:params:oauth:token-type:refresh_token' : 'urn:ietf:params:oauth:token-type:access_token';
+		$exchangeResponse = $this->requestTokenExchange($subjectToken, $subjectTokenType, $this->getClientID());
+
+		if (isset($exchangeResponse->error)) {
+			if (isset($exchangeResponse->error_description)) {
+				throw new OpenIDConnectClientException('TokenExchange response: ' . $exchangeResponse->error_description);
```

Comment:
error property should be part of the exception message as well

Reply:
but you will never get there - see https://github.com/JuliusPC/OpenID-Connect-PHP/blob/1a41cad55fb293590331758119296c2264b564d4/src/Client.php#L1124